src/anyio/_core/_subprocesses.py

```diff
@@ -65,20 +65,16 @@ async def drain_stream(stream: AsyncIterable[bytes], index: int) -> None:
         start_new_session=start_new_session,
     ) as process:
         stream_contents: list[bytes | None] = [None, None]
-        try:
-            async with create_task_group() as tg:
-                if process.stdout:
-                    tg.start_soon(drain_stream, process.stdout, 0)
-                if process.stderr:
-                    tg.start_soon(drain_stream, process.stderr, 1)
-                if process.stdin and input:
-                    await process.stdin.send(input)
-                    await process.stdin.aclose()
-
-                await process.wait()
-        except BaseException:
-            process.kill()
-            raise
+        async with create_task_group() as tg:
+            if process.stdout:
+                tg.start_soon(drain_stream, process.stdout, 0)
+            if process.stderr:
+                tg.start_soon(drain_stream, process.stderr, 1)
+            if process.stdin and input:
+                await process.stdin.send(input)
+                await process.stdin.aclose()
```

Comment:
Since we're modifying this code, let's add blank lines between control blocks.
```suggestion
            if process.stdout:
                tg.start_soon(drain_stream, process.stdout, 0)

            if process.stderr:
                tg.start_soon(drain_stream, process.stderr, 1)

            if process.stdin and input:
                await process.stdin.send(input)
                await process.stdin.aclose()
```